website/server/controllers/api-v3/user.js

```diff
@@ -1727,17 +1727,25 @@ api.movePinnedItem = {
       user.pinnedItemsOrder = currentPinnedItems.map(item => item.path);
     }
 
+    const itemExistInPinnedArray = user.pinnedItems.findIndex(item => item.path === path);
+
+    if (itemExistInPinnedArray === -1) {
+      throw new BadRequest(res.t('wrongItemPath', { path }, req.language));
```

Comment:
Will this cause a bug when a player tries to move one of the automatically pinned items that isn't in their `pinnedItems` array?

Reply:
Oh I thought this had been merged but given this doubt let's wait for @negue

Reply:
good point! gonna rework this a bit and try to add test for that as well - cause yes at that point that item isn't in user list (atleast not in the database array) the result however has those items .... so lets see how do it properly ^^

Reply:
the new push should allow to move official items too

Reply:
or not... `import getOfficialPinnedItems from '../../../common/script/libs/getOfficialPinnedItems';` can't be used in the user controller ..? any workaround to get this there too ?